### ISSUE TITLE
Using the VS Performance Profiler on a WASDK / WinUI 3 packaged desktop app always fails.

### ISSUE THREAD

**Issue by kmgallahan** [microsoft/WindowsAppSDK#1927]
### Describe the bug

Any time I try to use Visual Studio's Performance Profiler on a WASDK / WinUI 3 packaged desktop app it always fails with this message in the output:

```
Failed to disable Process Lifecycle Management (PLM) ...
```

The only search result is this Developer Community link that was reported and fixed in 2018:

https://developercommunity.visualstudio.com/t/failed-to-disable-process-lifecycle-management/249868

### Steps to reproduce the bug

1. Create a WinUI 3 single project desktop app using the built-in template
2. Debug > Performance Profiler...

### Expected behavior

Should be able to use the VS Performance Profiler.

### NuGet package version

1.0.0

### Packaging type

Packaged (MSIX)

### Windows version

Windows 10 version 21H2 (19044, November 2021 Update)

### IDE

Visual Studio 2022-preview

**Comment by DrusTheAxe** [comment]
You need to disable PLM so Windows doesn't try to suspend/etc your app while you're debugging. [Use PLMDebug](https://docs.microsoft.com/windows/uwp/debug-test-perf/testing-debugging-plm) to do this

**Comment by kmgallahan** [comment]
@DrusTheAxe Thanks for the reply.

I attempted to use PLMDebug:

```
C:\Program Files (x86)\Windows Kits\10\Debuggers\x64>plmdebug /enableDebug a7a4b0bd-2013-44df-bf14-5b977a586a76_1.0.0.0_x86__1geth39zw2vs4
Package full name is a7a4b0bd-2013-44df-bf14-5b977a586a76_1.0.0.0_x86__1geth39zw2vs4.
Enable debug mode
SUCCEEDED
```

However it still results in this:

```
Failed to disable Process Lifecycle Management (PLM) for 'a7a4b0bd-2013-44df-bf14-5b977a586a76_1.0.0.0_x86__1geth39zw2vs4' (0xe1110036).
Profiling of 'App6' started.
```

And VS just sits displaying:

![Screenshot 2021-12-28 182716](https://user-images.githubusercontent.com/5379488/147615397-8937ce99-301a-4645-a6b4-237b1dd667fa.jpg)

or

![Screenshot 2021-12-28 182734](https://user-images.githubusercontent.com/5379488/147615402-cf7c324e-8d58-493e-aa3d-cbff09a43552.jpg)

---

Is disabling PLM a one-time activity per-package name? Do I need to get involved with the part of the instructions referencing VSJITDebugger.exe?

Sorry for the trouble, none of this was necessary when profiling UWP apps. Is this a temporary process that won't be necessary in the future, or is manually disabling PLM per-package name just going to be something everyone has to figure out how to do if they want to use the profiler?

**Comment by chausner** [comment]
I am experiencing the same issue on a WinUI 3 app. Using Microsoft Visual Studio Community 2022 (64-Bit) 17.0.5.

**Comment by chausner** [comment]
A workaround that might work for some is to change the debug target in Performance Profiler to "Attach to a running app" or "Start an installed app". The problem does not occur for me there. However, not all of the tools in Performance Profiler are supported in those modes.

**Comment by Scottj1s** [comment]
@chausner, yes, I've discovered this works for both single-project and wapproj-based solutions, and keeps the .NET Object Allocations tool enabled (which is what I was after):

Choose Target:
Installed App…
Launch an installed Windows Store App

**Comment by Scottj1s** [comment]
I've opened a new issue with Visual Studio, as the Centennial detection logic appears not to be covering Windows App SDK apps:
https://developercommunity.visualstudio.com/t/performance-profiler-fails-with-windows-app-sdk-ap/1663956?from=email

**Comment by Scottj1s** [comment]
VS just reported a fix for this:
[Performance Profiler fails with Windows App SDK app: "Failed to disable Process Lifecycle Management (PLM)" - Visual Studio Feedback](https://developercommunity.visualstudio.com/t/performance-profiler-fails-with-windows-app-sdk-ap/1663956?from=email)

This may be available in 17.4 Preview 3 today (unclear)
